subsys/bluetooth/host/l2cap.c

```diff
@@ -1978,10 +1986,17 @@ static void le_ecred_conn_rsp(struct bt_l2cap *l2cap, uint8_t ident,
 
 			LOG_DBG("dcid 0x%04x", dcid);
 
-			/* If a Destination CID is 0x0000, the channel was not
+			/* Validate parameters before assignment. Requirements are from Core Spec
+			 * v6.0, Vol 3.A.4.26. If a Destination CID is 0x0000, the channel was not
 			 * established.
 			 */
-			if (!dcid) {
+			if (!dcid || !L2CAP_LE_CID_IS_DYN(dcid) ||
+			    !IN_RANGE(mtu, BT_L2CAP_ECRED_MIN_MTU, UINT16_MAX) ||
+			    !IN_RANGE(mps, BT_L2CAP_ECRED_MIN_MPS, BT_L2CAP_MAX_MPS) ||
+			    !IN_RANGE(credits, 1, UINT16_MAX)) {
```

Comment:
Suggest to add `#define`s for the minimum and maximum credit values too (can be in an internal file if the application doesn't need to care about these values)